mne/simulation/evoked.py

```diff
@@ -69,8 +69,14 @@ def simulate_evoked(fwd, stc, info, cov, snr=3., tmin=None, tmax=None,
     evoked = apply_forward(fwd, stc, info)
     if snr < np.inf:
         noise = simulate_noise_evoked(evoked, cov, iir_filter, random_state)
-        evoked_noise = add_noise_evoked(evoked, noise, snr,
-                                        tmin=tmin, tmax=tmax)
+        if evoked.data.shape != noise.data.shape:
```

Comment:
In ``simulate_noise_evoked`` we essentially just do
```
noise = evoked.copy()
noise.data = noise_data
```
, so the channels are actually the same anyway. Maybe this needs a more thorough look.